src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs

```diff
@@ -14,17 +14,17 @@
 using Microsoft.CodeAnalysis.Internal.Log;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Remote;
-using Microsoft.CodeAnalysis.Rename.ConflictEngine;
-using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Rename
 {
     /// <summary>
-    /// Holds the Locations of a symbol that should be renamed, along with the symbol and Solution
-    /// for the set.
+    /// Holds the Locations of a symbol that should be renamed, along with the symbol and Solution for the set. It is
+    /// considered 'heavy weight' because it holds onto large entities (like Symbols) and thus should not be marshaled
+    /// to/from a host to OOP.
```

Comment:
note: no marshalling code actually exists for this type.  this is intentional to prevent that from accidentally happening.